reduce/index.js

```diff
@@ -1,26 +1,25 @@
 
 function sumNumbers(array) {
-  // TODO: REPLACE THIS WITH A .REDUCE
-  let sum = 0;
-  for (let i = 0; i < array.length; i++) {
-    sum = sum + array[i]
-  }
-  return array
+  return array.reduce( (sum, value) => sum + value)
 }
 
 // 'Transform' an Array to Another Size Array w/ .reduce
-function squareEvenNumbers(array) {
-  // TODO: Use at least 1 .reduce,
-  // square ONLY the even numbers
-  // & push the results into the return array
-  return
-
+function squareEvenNumbers(values) {
+  return values.filter( (value) => value % 2 === 0).reduce( (total, amount) => {
+    total.push(amount * amount)
+    return total
+  }, [])
 }
-
 // Use reduce to group an array of
 // objects on a specified key
 function groupByKey(array, key = 'key') {
-  return
+  return array.reduce( (acc, cur) => {
+    if (acc[cur.key] === undefined) {
+      acc[cur.key] = [] 
+    }
+    acc[cur.key].push(cur)
```

Comment:
Update the`acc[cur.key]` to be `acc[key]` so it uses the 2nd argument passed into the `groupByKey` function.

👍